crates/matrix-sdk/src/send_queue/mod.rs

```diff
@@ -724,13 +769,180 @@ impl RoomSendQueue {
         info!("exited sending task");
     }
 
+    /// Try to create metadata required to compute the progress of a media
+    /// upload.
+    async fn try_create_media_upload_info(
+        queued_request: &QueuedRequest,
+        room: &Room,
+        queue: &QueueStorage,
+    ) -> Option<MediaUploadInfo> {
+        let QueuedRequestKind::MediaUpload {
+            cache_key,
+            thumbnail_source,
+            related_to,
+            #[cfg(feature = "unstable-msc4274")]
+            accumulated,
+            ..
+        } = &queued_request.kind
+        else {
+            return None;
+        };
+
+        // Determine the item's index, if this is a gallery upload.
+        let index = {
+            cfg_if::cfg_if! {
+                if #[cfg(feature = "unstable-msc4274")] {
+                    accumulated.len()
+                } else {
+                    0 // Before MSC4274 only a single file (and thumbnail) could be sent per event.
+                }
+            }
+        };
+
+        // Get the size of the file being uploaded from the event cache.
+        let bytes = if let Ok(cache) = room.client().event_cache_store().lock().await {
+            if let Ok(Some(content)) = cache.get_media_content(cache_key).await {
```

Comment:
We really can't just ignore errors, as they're unexpected. We need proper error handling in this function.